### PR TITLE
fix(dagster/decorators): op_decorator documentation update

### PR DESCRIPTION
**Pull request by mict0**
Updating legacy return types from Solid to OpDefintion

### Summary & Motivation
It's weird to see legacy types for newcomers.

### CODE REVIEW

File: python_modules/dagster/dagster/core/definitions/decorators/op_decorator.py [hunk 1]
```diff
@@ -181,7 +181,7 @@ def _resolve_output_defs_from_outs(
 
 
 @overload
-def op(name: Callable[..., Any]) -> SolidDefinition:
+def op(name: Callable[..., Any]) -> OpDefinition:
```
Comment (dpeng817) [inline python_modules/dagster/dagster/core/definitions/decorators/op_decorator.py:184]: When using a type annotation imported as a future, you need to use quotations around the annotation:
```suggestion
def op(name: Callable[..., Any]) -> "OpDefinition":
```

File: python_modules/dagster/dagster/core/definitions/decorators/op_decorator.py [hunk 2]
```diff
@@ -214,7 +214,7 @@ def op(
     retry_policy: Optional[RetryPolicy] = None,
     input_defs: Optional[List[InputDefinition]] = None,
     output_defs: Optional[List[OutputDefinition]] = None,
-) -> Union[SolidDefinition, _Op]:
+) -> Union[OpDefinition, _Op]:
```
Comment (dpeng817) [inline python_modules/dagster/dagster/core/definitions/decorators/op_decorator.py:217]: Same as [1]:
```suggestion
) -> Union["OpDefinition", _Op]:
```

**Review by dpeng817 — CHANGES_REQUESTED** [review]
Thanks for doing this! A simple change and this should be good to go.

**Comment by dpeng817** [comment]
@mict0 only one more small change, just need to get rid of the import of `SolidDefinition` in `op_decorator.py` now that it's no longer used :)

**Comment by mict0** [comment]
> @mict0 only one more small change, just need to get rid of the import of `SolidDefinition` in `op_decorator.py` now that it's no longer used :)

Actually it's used in line 8 `class OpDefinition(SolidDefinition):` since OpDefinition is inheriting from SolidDefinition, not sure if bigger change is needed. I think you'd like to keep SolidDefinition for legacy code, but maybe should create separate class OpDefinition for the future use-cases

**Comment by dpeng817** [comment]
@mict0 the line you're referring to is within `op_definition.py`, but the proposed changes are in `op_decorator.py`. The only place the `SolidDefinition` import was used within that file was these annotations, so it's necessary to get rid of the import in that file.

**Comment by mict0** [comment]
> @mict0 the line you're referring to is within `op_definition.py`, but the proposed changes are in `op_decorator.py`. The only place the `SolidDefinition` import was used within that file was these annotations, so it's necessary to get rid of the import in that file.

oh, you're right my bad. changes added, import is removed

**Review by dpeng817 — APPROVED** [review]

**Comment by dpeng817** [comment]
Thanks for your contribution @mict0 ! This will go into the next release, and we'll shout it out in the changelog :)

**Pull request merged**